feature-nft-impl/src/main/java/io/novafoundation/nova/feature_nft_impl/presentation/nft/send/confirm/ConfirmNftSendViewModel.kt

```diff
@@ -0,0 +1,194 @@
+        .map { it.token }
+        .inBackground()
+        .share()
+
+    val originFeeMixin: FeeLoaderMixin.Presentation = feeLoaderMixinFactory.create(commissionTokenFlow)
+
+    private val currentAccount = selectedAccountUseCase.selectedMetaAccountFlow()
+        .inBackground()
+        .share()
+
+    val recipientModel = flowOf {
+        createAddressModel(
+            address = transferDraft.recipientAddress,
+            chain = originChain(),
+            resolveName = true
+        )
+    }
+        .inBackground()
+        .share()
+
+    val senderModel = currentAccount.mapLatest { metaAccount ->
+        createAddressModel(
+            address = metaAccount.requireAddressIn(originChain()),
+            chain = originChain(),
+            resolveName = false
+        )
+    }
+        .inBackground()
+        .share()
+
+    val originChainUi = flowOf { originChain() }
+        .map(::mapChainToUi)
+        .shareInBackground()
+
+    val wallet = walletUiUseCase.selectedWalletUiFlow()
+        .inBackground()
+        .share()
+
+    private val _transferSubmittingLiveData = MutableStateFlow(false)
+
+    val sendButtonStateLiveData = _transferSubmittingLiveData.map { submitting ->
+        if (submitting) {
+            ButtonState.PROGRESS
+        } else {
+            ButtonState.NORMAL
+        }
+    }
+
+    init {
+        setInitialState()
+    }
+
+    fun backClicked() {
+        router.back()
+    }
+
+    fun recipientAddressClicked() = launch {
+        showExternalActions(transferDraft.recipientAddress, originChain)
+    }
+
+    fun senderAddressClicked() = launch {
+        showExternalActions(senderModel.first().address, originChain)
+    }
+
+    private suspend fun showExternalActions(
+        address: String,
+        chain: Deferred<Chain>,
+    ) {
+        externalActions.showExternalActions(ExternalActions.Type.Address(address), chain())
+    }
+
+    fun submitClicked() = launch {
+        val payload = buildValidationPayload()
+        validationExecutor.requireValid(
+            validationSystem = nftSendInteractor.defaultValidationSystem(),
+            payload = payload,
+            progressConsumer = _transferSubmittingLiveData.progressConsumer(),
+            validationFailureTransformer = { mapNftTransferValidationFailureToUI(resourceManager, it) }
+        ) {
+            performTransfer(it.transfer)
+        }
+    }
+
+    private fun setInitialState() = launch {
+        originFeeMixin.setFee(transferDraft.originFee)
+    }
+
+    private suspend fun createAddressModel(
+        address: String,
+        chain: Chain,
+        resolveName: Boolean
+    ) =
+        addressIconGenerator.createAddressModel(
+            chain = chain,
+            sizeInDp = AddressIconGenerator.SIZE_MEDIUM,
+            address = address,
+            background = AddressIconGenerator.BACKGROUND_TRANSPARENT,
+            addressDisplayUseCase = addressDisplayUseCase.takeIf { resolveName }
```

Comment:
Consider using existing `createAccountAddressModel` extension

Reply:
I can't use this extension due to specific logic in:
1) `sizeInDp = AddressIconGenerator.SIZE_MEDIUM,`
2) `addressDisplayUseCase = addressDisplayUseCase.takeIf { resolveName }`